server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/ConsumerOptions.java

```diff
@@ -0,0 +1,42 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.jms;
+
+import org.apache.james.queue.api.MailQueue;
+
+/**
+ * Provides additional options when creating consumers.
+ */
+public interface ConsumerOptions {
+    /**
+     * The empty consumer options.
+     */
+    static ConsumerOptions empty() {
+        return queueName -> queueName;
+    }
+
+    /**
+     * The queue name with options for performing {@link MailQueue#deQueue()} operation. The implementation may
+     * return {@code queueName} itself if it does not support any options.
+     *
+     * @return The queue name maybe with additional options.
+     */
+    String applyForDequeue(String queueName);
```

Comment:
maybe you already discussed that with @chibenwa but this API makes the assumption that the queue name is actually an url where you can encode options through queryparams.
I should clearly hide that fact because it's more than likely no useful for some kinds of MailQueue.
What about not putting this method only in implementations that make sense ?

Reply:
Our issue basically is related to `JMSMailQueue` from with we have reference on `ConsumerOptions` and not `ActiveMQConsumerOptions`. Although we can do `instanceof ActiveMQConsumerOptions` and then call `applyForDequeue`. That `instanceof` check will be a bit ugly in my opinion.

Reply:
Unfortunately `queue-jms` does not depend on `queue-activemq` which is logical. Therefore previous comment cannot be implemented...